### PR TITLE
Clean up; Implement forgive all feature #5, #7

### LINKED ISSUE #5: Code improvements for prank/forgive
https://github.com/morganlutz/robin-bot/blob/master/app.js#L100

Consider using a switch statement and checking for `pranked` instead of `splitMsg[1]` just once.  

e.g.:

``` javascript
if (pranked) {
  switch(splitMsg[0]) {
  case 'PRANK':
    …stuff…
    break;
  }
  …more cases…
  default:
    …"Holy Try again, that I don't know what to do with the command splitMsg[0]…"
} else {
  …"Holy Try again, send an empty message…"
}
```

### PR DESCRIPTION
**Pull request by morganlutz**
Resolves #5 #7

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -97,33 +97,51 @@ controller.on('slash_command', (bot, message) => {
...
+          } else {
+            reply.private = 'Holy Prankster! Only the person that pranked you can remove you from my list!';
+          }
+        } else if(pranked === 'all') {
+          let forgiveAllReply = 'Holy Holiness! Users removed from the prank list: \n';
+          console.log('ALL_PRANKED', RobinBot.prankedUsers);
+          for(let [prankedName, prankedObject] of Object.entries(RobinBot.prankedUsers)) {
+            if(message.user === prankedObject.prankerID) {
+              forgiveAllReply += prankedName + ' \n';
+              delete RobinBot.prankedUsers[prankedName];
```
Comment (infernoboy) [inline app.js:129]: Is it safe to delete from an object you are iterating over after using Object.entries()?

File: app.js [hunk 2]
```diff
@@ -97,33 +97,51 @@ controller.on('slash_command', (bot, message) => {
...
+            delete RobinBot.prankedUsers[pranked];
+          } else {
+            reply.private = 'Holy Prankster! Only the person that pranked you can remove you from my list!';
+          }
+        } else if(pranked === 'all') {
+          let forgiveAllReply = 'Holy Holiness! Users removed from the prank list: \n';
+          console.log('ALL_PRANKED', RobinBot.prankedUsers);
+          for(let [prankedName, prankedObject] of Object.entries(RobinBot.prankedUsers)) {
+            if(message.user === prankedObject.prankerID) {
+              forgiveAllReply += prankedName + ' \n';
```
Comment (infernoboy) [inline app.js]: I think this is better off in a single line. You can create an empty array above `let forgiveAllReply` and then push into it all the names that are being removed. Then you can append the `forgiveAllReply` with `arrayOfPrankedUsersRemoved.join(',')`.

File: app.js [hunk 3]
```diff
@@ -97,33 +97,51 @@ controller.on('slash_command', (bot, message) => {
...
-          lastPrankTime: 0,
-          nextPrankTime: 0,
-          prankerID: message.user,
-          lastPrankMessage: ''
-        };
-      } else {
-        reply.private = `Holy Confusion! ${pranked} is already on my list!`;
+    if(pranked) {
+      switch (splitMsg[0]) {
+      case 'PRANK': {
```
Comment (infernoboy) [inline app.js]: You shouldn't be wrapping your cases in `{ }`!

File: botKitHelper.js [hunk 4]
```diff
@@ -32,7 +32,7 @@ const BotKitHelper =  {
       return filteredMembers;
 
     }, (err) => console.error(err));
-  }
+  },
```
Comment (infernoboy) [inline botKitHelper.js]: Avoid trailing commas.

File: app.js [hunk 5]
```diff
@@ -97,33 +97,51 @@ controller.on('slash_command', (bot, message) => {
...
+        if(RobinBot.prankedUsers.hasOwnProperty(pranked)) {
+          if(message.user === RobinBot.prankedUsers[pranked].prankerID) {
+            reply.private = `Holy Prankster! ${pranked} has been removed from my list!`;
+            delete RobinBot.prankedUsers[pranked];
+          } else {
+            reply.private = 'Holy Prankster! Only the person that pranked you can remove you from my list!';
+          }
+        } else if(pranked === 'all') {
+          let forgiveAllReply = 'Holy Holiness! Users removed from the prank list: \n';
+          console.log('ALL_PRANKED', RobinBot.prankedUsers);
```
Comment (infernoboy) [inline app.js]: Don't commit with code intended for debugging purposes!

File: app.js (hunk 3 above)
Comment (morganlutz) [reply app.js]: I wrapped my cases because the linter yelled at me!

File: app.js (hunk 1 above)
Comment (morganlutz) [reply app.js:129]: I dunno. Is it??

File: app.js (hunk 2 above)
Comment (morganlutz) [reply app.js]: Fixed in 2a979f796

File: app.js (hunk 3 above)
Comment (morganlutz) [reply app.js]: Updated in 15dded6ac

File: botKitHelper.js (hunk 4 above)
Comment (morganlutz) [reply botKitHelper.js]: Fixed in 9aa5e151

File: app.js (hunk 1 above)
Comment (infernoboy) [reply app.js:129]: Why yes, yes it is!

File: app.js (hunk 5 above)
Comment (morganlutz) [reply app.js]: I'll remove all logs before tagging

**Pull request merged**